File: Week 1/Python Codes/SortAnArrayOf012.py
#code
def sort_array(arr):
    count_0 = 0
    count_1 = 0
    count_2 = 0
    for num in arr:
        if num == 0:
            count_0 += 1
        if num == 1:
            count_1 += 1
        if num == 2:
            count_2 += 1
    return [0]*count_0 + [1]*count_1 + [2]*count_2

if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        n = int(input())
        arr = [int(x) for x in input().split()]
        result = sort_array(arr)
        for num in result:
            print(num, end = " ")
        print()

File: Week 1/C++ Codes/SortAnArrayOf012.cpp
#include <iostream>
using namespace std;

void sort_array(int arr[], int n) {
    int count_0 = 0, count_1 = 0, count_2 = 0, i;
    for (i = 0; i < n; i++) {
        if (arr[i] == 0)
            count_0 += 1;
        if (arr[i] == 1)
            count_1 += 1;
        if (arr[i] == 2)
            count_2 += 1;
    }
    int result[n];
    i = 0;
    while (count_0 > 0) {
        result[i] = 0;
        count_0 -= 1;
        i += 1;
    }
    while (count_1 > 0) {
        result[i] = 1;
        count_1 -= 1;
        i += 1;
    }
    while (count_2 > 0) {
        result[i] = 2;
        count_2 -= 1;
        i += 1;
    }
    for (int i = 0; i < n; i++) {
        cout<<result[i]<<" ";
    }
    cout<<"\n";
}

int main() {
	//code
	int t, n;
	cin>>t;
	for(int i = 0; i < t; i++) {
        cin>>n;
        int arr[n];
        for (int j = 0; j < n; j++) {
            cin>>arr[j];
        }
        sort_array(arr, n);
	}
	return 0;
}
